### PR TITLE
Order filters: custom date range

### LINKED ISSUE #4934: 📦  Order List filters
Enable filtering orders by status or date range. This task also included a new design for the screen were tabs will be removed and a single order list will be displayed. 

More details p91TBi-5L9-p2

### 💻  Tasks

- [x] Enable filtering by order status: 
       - Part I:  #5039
       - Part II  #5087
- [x] Add filter for given ranges: Today, Last 2 Days, This Week, This Month: 
      - WooCommerce changes #5104 
      - FluxC changes [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2167)
- [x] Split order filters view model into 2 independent view models and properly handle state recovery on activity recreation: #5140
- [x] Save selected filters on disk to be kept across app sessions: #5169

To be done last 
- [x] Define and implement tracking: #5220
- [x] Have a single order list, remove tabs and apply always any filter selection to order list: #5219 
- [x] Remove status filters from search view and fix UI glitches between search feature and new filters button: #5239

Once the previous tasks are done, review if we still want to add this. 

- [x] Add filter for custom date range: #5260

### Design overview

Screenshot 2021-10-05 at 09 52 13

Screenshot 2021-10-05 at 13 02 19

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Closes: #4934 

### Description
Enable filtering orders by date range. There were 2 possible approaches according to designs: 

Screenshot 2021-11-17 at 19 14 22

I tested both. Finally opted for using the [MateriaDateRangePicker](https://material.io/components/date-pickers/android#date-range-pickers) because:
- It provides a more consistent experience with other apps
- Remove an unnecessary step on the filters flow
- It provides date range validation out of the box, for example making sure startDate < endDate
- Way less code (I was able to remove Fragment and ViewModel specifically added for handling date ranges)

This should be validated by @adamzelinski, so please take a look at the screen recordings when possible, as maybe we want to fine tune some colors from the picker. I have also pasted a screenshow for dark mode. 

### Testing instructions
- Open filters
- Select "Date Range" 
- Select "Custom Range" 
- Play selecting different ranges

### Images/gif

https://user-images.githubusercontent.com/2663464/142257626-2b432a8b-90a3-4629-ac35-3f6af4cc4625.mp4

Dark mode date range picker: 

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt [hunk 1]
```diff
@@ -7,10 +7,33 @@ import android.content.Context
 import android.content.SharedPreferences
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt:7]: Most changes on this file are irrelevant. Just used fully qualified import for each key (I shouldn't have pressed alt + enter 😅 )

I'll highlight the relevant changes from file.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt [hunk 2]
```diff
@@ -457,12 +482,39 @@ object AppPrefs {
...
+            "$ORDER_FILTER_CUSTOM_DATE_RANGE_START:$selectedSiteId",
+            startDateMillis
+        )
+        PreferenceUtils.setLong(
+            getPreferences(),
+            "$ORDER_FILTER_CUSTOM_DATE_RANGE_END:$selectedSiteId",
+            endDateMillis
+        )
+    }
+
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/AppPrefs.kt:511]: New code to save and retrieve date range from shared prefs

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt [hunk 3]
```diff
@@ -147,8 +148,8 @@ class AnalyticsTracker private constructor(private val context: Context) {
         ORDERS_LIST_LOADED,
         ORDERS_LIST_SHARE_YOUR_STORE_BUTTON_TAPPED,
         ORDERS_LIST_PULLED_TO_REFRESH,
-        ORDERS_LIST_MENU_FILTER_TAPPED,
         ORDERS_LIST_MENU_SEARCH_TAPPED,
+        ORDERS_LIST_VIEW_FILTER_OPTIONS_TAPPED,
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt:152]: Just adding a missing event that needs to be tracked when filters button is tapped from order list screen. The event is tracked [here](https://github.com/woocommerce/woocommerce-android/pull/5260/files#diff-daf78e7103c7d0e51140c59b8490193f96db648214827d13e0d3f66c42c576d9L367)

**Review by 0nko — APPROVED** [review]
Works & looks great! 👍

**Comment by JorgeMucientes** [comment]
Hey @adamzelinski 👋🏼   Could you please make a quick design review of the feature before I merge this? :)

**Comment by ghost** [comment]
Hey, @JorgeMucientes this is looking really great thanks I'll approve! One small thing I noticed which may not be a part of this pr is that when you select a date range the dates aren't shown in the order list like so.
Screen Shot 2021-11-23 at 11 21 07 am

**Review by ghost — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Thanks for the review @adamzelinski. Good catch I totally missed to do that. I will address that here #5313 and proceed to merge this.

**Pull request merged**